internal/core/config/monitor.go

```diff
@@ -51,6 +51,10 @@ type MonitorConfig struct {
 	// is useful when you have an endpoint whose identity is not particularly
 	// important since it acts largely as a proxy or adapter for other metrics.
 	DisableEndpointDimensions bool `yaml:"disableEndpointDimensions" json:"disableEndpointDimensions"`
+	// Whether it will send all custom and non-basic metrics as well.
+	// For the monitors which utilize this config, a list of basic metrics will be managed within the
+	// monitor module.
+	EnhancedMetrics bool `yaml:"enhancedMetrics" json:"enhancedMetrics" default:"false"`
```

Comment:
This is too generic of a place to put this.  It definitely doesn't apply to all monitors.  What I was saying about this struct is that it is embedded, just like you could also do with another struct that is common to ECS and Docker monitors that contains the shared enhanced metrics flags.  Like this:

```
type EnhancedMetricsConfig struct {
    EnableExtraBlockIOMetrics bool `yaml:"enableExtraBlockIOMetrics"`
   ...
}
```

That could be used like:

```
type Config struct{
    config.MonitorConfig
    docker.EnhancedMetricsConfig
...
}
```

Reply:
That makes sense.